### PR TITLE
Add signature removal to tag-docker operations

### PR DESCRIPTION
**Pull request by querti**
A somewhat unrelated change is adding the filtering of claim messages
which already exist as signatures.

### CODE REVIEW

**Review by midnightercz — APPROVED** [review]

File: pubtools/_quay/signature_remover.py [hunk 1]
```diff
@@ -249,3 +249,80 @@ def remove_repository_signatures(
...
+            else []
+        )
+
+        remove_signature_ids = []
+        for sig in self.get_signatures_from_pyxis(
+            image_digests, pyxis_server, pyxis_krb_principal, pyxis_krb_ktfile
+        ):
+            # if signature corresponds to to-be-removed digest+reference and isn't among new sigs
+            if (
+                sig["manifest_digest"] in image_digests
```
[the remark this answers is not in the document]
Comment (querti) [reply pubtools/_quay/signature_remover.py:315]: I suppose not, but I guess the check is a bit more demonstrative in terms of what it's trying to accomplish

File: pubtools/_quay/tag_docker.py [hunk 2]
```diff
@@ -579,7 +595,17 @@ def merge_manifest_lists_sign_images(self, push_item, tag, add_archs, signature_
...
+        self.signature_remover.remove_tag_signatures(
+            reference=dest_image,
+            pyxis_server=self.target_settings["pyxis_server"],
+            pyxis_krb_principal=self.target_settings["iib_krb_principal"],
+            pyxis_krb_ktfile=self.target_settings.get("iib_krb_ktfile", None),
+            exclude_by_claims=claim_messages,
+        )
+
+        filtered_claim_messages = signature_handler.filter_claim_messages(claim_messages)
+        signature_handler.sign_claim_messages(filtered_claim_messages, True, True)
```
Comment (emilyzheng) [inline pubtools/_quay/tag_docker.py:608]: The second 'True' is to enable filter_claim_messages.

File: pubtools/_quay/signature_remover.py [hunk 3]
```diff
@@ -249,3 +249,80 @@ def remove_repository_signatures(
...
+                and sig["reference"] == reference
```
[the remark this answers is not in the document]
Comment (querti) [reply pubtools/_quay/signature_remover.py]: Good catch, fixed

File: pubtools/_quay/tag_docker.py (hunk 2 above)
Comment (querti) [reply pubtools/_quay/tag_docker.py]: :man_facepalming: :man_facepalming: :man_facepalming: fixed

File: pubtools/_quay/signature_remover.py [hunk 4]
```diff
@@ -249,3 +253,82 @@ def remove_repository_signatures(
...
+        """
+        if "@" in reference:
+            raise ValueError("Image, whose signatures are being removed must be specified by tag.")
+
+        full_repo, tag = reference.split(":", 1)
+        external_repo = get_external_container_repo_name(full_repo.split("/")[-1])
+        image_digests = []
+        repo_data = self.quay_api_client.get_repository_data(full_repo.split("/", 1)[-1])
+        # if image_id of the tag is specified, the image is V2S2 AKA source image
+        if repo_data["tags"][tag]["image_id"]:
```
Comment (emilyzheng) [inline pubtools/_quay/signature_remover.py:298]: Getting image_digests should be skipped when reference is a not existing image.

Comment (querti) [reply pubtools/_quay/signature_remover.py]: I skipped the rest of the method if the image doesn't exist

**Review by emilyzheng — APPROVED** [review]

**Pull request merged**